planner/core/logical_plan_builder.go

```diff
@@ -3223,9 +3223,42 @@ func (b *PlanBuilder) buildUpdate(ctx context.Context, update *ast.UpdateStmt) (
 		tblID2table[id], _ = b.is.TableByID(id)
 	}
 	updt.TblColPosInfos, err = buildColumns2Handle(updt.OutputNames(), tblID2Handle, tblID2table, true)
+	if err == nil {
```

Comment:
```suggestion
	if err != nil {
```
Better to use this style.

Reply:
No, this logic is wrong